src/volatile_set.c

```diff
@@ -0,0 +1,79 @@
+#include <string.h>
+#include "volatile_set.h"
+#include "zmalloc.h"
+#include "config.h"
+#include "endianconv.h"
+#include "serverassert.h"
+
+#define EXPIRY_HASH_SIZE 16
+volatile_set *createVolatileSet(volatileEntryType *type) {
```

Comment:
I noticed these two files volatile_set.{c,h} have zero comments. Postponing code comments is not nice for reviewers. :stuck_out_tongue_closed_eyes: 
 
I suggest writing some quick comments (and unit tests) first, even before the actual implementation. It can help when designing the API and reasoning about invariants and pre-conditions.

Reply:
O.K - the current volatile_set was only added as a reference and a way to quickly implement the AOF rewrite. @xbasel will soon issue a PR for active expiration support which will introduce a refactor implementation of the volatile_set

Reply:
I was more thinking about a comment about the high-level idea. What is a volatile set at all? A set of ...

Reply:
Sure. will do it